crates/xilem_core/src/vec_splice.rs

```diff
@@ -51,11 +51,11 @@ impl<'a, 'b, T> VecSplice<'a, 'b, T> {
     }
 
     pub fn peek(&self) -> Option<&T> {
-        self.v.last()
+        self.v.get(self.ix)
```

Comment:
Ah, just found this locally too :grinning:  thanks!

I think this should be `ix - 1` though, since we add 1 to `ix` in mutate

Reply:
Maybe peek isn't the best name for this method either, as what it's used for is look at the last element

Reply:
> I think this should be `ix - 1` though, since we add 1 to `ix` in mutate

You're right, just checked it again, `ix` is incremented after accessing it.

> Maybe peek isn't the best name for this method either

You're probably right. Maybe call it `current` instead? I mean `peek` is not totally inaccurate, as it should return the last modified element (if not yet clear, `VecSplice` sweeps through the elements). But I see that it may confuse (additionally to the already quite complex code of `VecSplice`)

Reply:
`current` feels better, but maybe something like `last_mutated` would be the most descriptive.

Reply:
Yes makes sense, I've renamed it to `last_mutated` and fixed the index access.